src/core/utils.js

```diff
@@ -751,19 +751,16 @@ export function getLinkTargets(elem) {
 export function renameElement(elem, newName) {
   if (elem.localName === newName) return elem;
   const newElement = elem.ownerDocument.createElement(newName);
-
+  if (!document.contains(elem)) throw new Error("Node not in document.");
   // copy attributes
-  for (let i = 0, n = elem.attributes.length; i < n; i++) {
-    const { name, value } = elem.attributes[i];
+  for (const attribute of [...elem.attributes]) {
+    const { name, value } = attribute;
     newElement.setAttribute(name, value);
   }
-
   // copy child nodes
-  do {
+  while (elem.firstChild) {
```

Comment:
Off-topic: Why does this loop - `while (elem.firstChild)` work? should it not be infinite loop? 👨‍🎓

Reply:
ah, because `newElement.appendChild(elem.firstChild);` steals `elem.firstChild `, so eventually `elem.firstChild` becomes `null`.

Reply:
steals? wow. never knew this. will read about it.
(gotta study for exam first 😆)

Reply:
yeah, it relocates it in the tree. Approve before you run off!!!!

Reply:
Test is failing though.

Reply:
Tests! ppfffff! SHIP IT!

Reply:
Oh, I'd assumed that nodes would have to be connected for it to work. But seems they don't, so removing.

Reply:
ok, fixed... tests should pass now.